pkgs/tools/networking/teavpn2/nix.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/src/teavpn2/net/linux/iface.c b/src/teavpn2/net/linux/iface.c
+index a77c1c8..e59c901 100644
+--- a/src/teavpn2/net/linux/iface.c
++++ b/src/teavpn2/net/linux/iface.c
+@@ -327,7 +327,8 @@ static __cold __always_inline const char *find_ip_cmd(void)
+ 		"/usr/sbin/ip",
+ 		"/usr/local/bin/ip",
+ 		"/usr/local/sbin/ip",
+-		"/data/data/com.termux/files/usr/bin/ip"
++		"/data/data/com.termux/files/usr/bin/ip",
++		"/run/current-system/sw/bin/ip"
```

Comment:
```suggestion
+		"${iproute2}/bin/ip"
```

Reply:
And regarding these changes, do I need to add iproute2 to the default.nix? because I got this error when I'm trying to run the package 
```
[Wed Aug  9 19:00:28 2023] Locating ${iproute2}/bin/ip: No such file or directory
```

Reply:
Yes, please do so.

Reply:
Done

Reply:
I still got the same error 
```
[Wed Aug  9 19:34:39 2023] Locating /bin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating /sbin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating /usr/bin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating /usr/sbin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating /usr/local/bin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating /usr/local/sbin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating /data/data/com.termux/files/usr/bin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Locating ${iproute2}/bin/ip: No such file or directory
[Wed Aug  9 19:34:39 2023] Error: Cannot find ip bin executable file
```

Reply:
Sorry, my bad.
We should use `substituteAll`.

```diff
- "${iproute2}/bin/ip"
+ "@iproute2@"/bin/ip"
```

and in default.nix,

```nix
  patches = [
    (substituteAll {
      src = ./nix.patch;
      inherit iproute2;
    })
  ];
```

Reply:
What do you think about this 
```
  postPatch = ''
   substituteInPlace src/teavpn2/net/linux/iface.c \
     --replace '"/data/data/com.termux/files/usr/bin/ip"' '"/data/data/com.termux/files/usr/bin/ip", "${iproute2}/bin/ip"'
  '';
```
So I don't need to use nix.patch anymore. I tested and it's build + running now.

Reply:
sounds good

Reply:
Tested both substituteAll and the postPatch, It's building and running.